### PR TITLE
premove survey changes

### PR DESCRIPTION
**Pull request by amitch23**
## Description

This PR makes the following changes based on this doc from design: https://docs.google.com/spreadsheets/d/1leyc06TJ5nGXKPIZXzZ4xLGIftRWFPr3MSKYfBVRQQ4/edit?usp=sharing

* removes `latest pickup date`
* removes `earliest delivery date`
* changes `latest delivery date` to `planned_delivery_date`
* changes pack_date and pickup_date to be prefixed with "planned"
* adds a field for how the PM survey was conducted - `pm_survey_method`. i.e. In person, over the phone, or video.

## Setup

-populate db with hhg awarded data: `make db_populate_e2e`
-run tsp app, select hhg move info from table - verify the values in the premove survey box match the data. Edit the values, ensure they persist.

## Code Review Verification Steps

* [ ] End to end tests pass (`make e2e_test`).
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/159965926) for this change
* Reference dates doc: https://docs.google.com/spreadsheets/d/1leyc06TJ5nGXKPIZXzZ4xLGIftRWFPr3MSKYfBVRQQ4/edit?usp=sharing

## Screenshots
screen shot 2018-08-23 at 5 40 11 pm
screen shot 2018-08-23 at 5 40 21 pm

### CODE REVIEW

File: pkg/handlers/shipments_public.go [hunk 1]
```diff
@@ -152,21 +151,31 @@ func (h PublicCreateShipmentRejectHandler) Handle(params publicshipmentop.Create
 
 func publicPatchShipmentWithPayload(shipment *models.Shipment, payload *apimessages.Shipment) {
 	// Premove Survey values entered by TSP agent
-	requiredValue := payload.PmSurveyPackDate
```
Comment (stangah) [inline pkg/handlers/shipments_public.go]: Did you mean to undo this? It becomes hard to delete optional fields from the survey form if we do our usual way of handling PATCH

File: pkg/testdatagen/scenario/e2ebasic.go [hunk 2]
```diff
@@ -208,14 +208,19 @@ func (e e2eBasicScenario) Run(db *pop.Connection, loader *uploader.Uploader) {
 
 	// Service member with uploaded orders and a new shipment move
 	email = "hhg@incomple.te"
+	uuidStr = "ebc176e0-bb34-47d4-ba37-ff13e2dd40b9"
 
-	hhg0 := testdatagen.MakeShipment(db, testdatagen.Assertions{
+	testdatagen.MakeUser(db, testdatagen.Assertions{
```
Comment (stangah) [inline pkg/testdatagen/scenario/e2ebasic.go]: Did you intentionally have `MakeUser` be its own call again, or is this from a merge conflict?

File: pkg/handlers/shipments_public.go (hunk 1 above)
Comment (tinyels) [reply pkg/handlers/shipments_public.go]: I think this change is fine. The api shouldn't assume that our interface is the only way it is getting called. Especially the public api.

File: pkg/testdatagen/scenario/e2ebasic.go (hunk 2 above)
Comment (tinyels) [reply pkg/testdatagen/scenario/e2ebasic.go]: Very happy this got fixed

**Comment by tinyels** [comment]
This isn't directly related to changes in this PR but `pkg/handlers/shipments_public.go` is erroring on line 80 with "sql: no rows in result set" if there are no shipments. I don't think this should error. I'm also not getting any shipments in the tsp view when I run `make db_populate_e2e`, which seems to be a change (and contradicts the instructions on this PR).

File: pkg/handlers/shipments_public.go (hunk 1 above)
Comment (stangah) [reply pkg/handlers/shipments_public.go]: I agree with that idea, but it still leaves users with no way to blank optional fields in the premove survey

Comment (amitch23) [reply pkg/handlers/shipments_public.go]: No, this was not intentional. I didn't realize this had been changed, and it must have been updated without a conflict when I merged your changes. I'll change it back to enable blank optional fields getting set.

File: pkg/testdatagen/scenario/e2ebasic.go (hunk 2 above)
Comment (amitch23) [reply pkg/testdatagen/scenario/e2ebasic.go]: Removed it.

**Pull request merged**